src/main/xerus/monstercat/downloader/Download.kt

```diff
@@ -81,10 +81,15 @@ class ReleaseDownload(private val release: Release, private var tracks: Collecti
 	
 	fun downloadTrack(releaseId: String, trackId: String, path: Path) {
 		val connection = APIConnection("api", "release", releaseId, "download").addQueries("method=download", "type=${QUALITY()}", "track=$trackId")
-		val entity = connection.getResponse().entity
+		val httpResponse = connection.getResponse()
+		val entity = httpResponse.entity
 		val contentLength = entity.contentLength
 		if(contentLength == 0L)
 			throw EmptyResponseException(connection.uri.toString())
+		if(!listOf("application/octet-stream", "audio/flac", "audio/wav", "audio/mpeg").contains(entity.contentType.value))
+			throw WrongResponseTypeException(connection.uri.toString(), entity.contentType.value)
+		if(httpResponse.statusLine.statusCode != 200)
```

Comment:
All 2xx codes indicate success, not just 200

Reply:
But some 2xx codes are "resource created" which could be harmful. 200 is THE "GOOD DOWNLOAD" one.